src/context/WalletProvider/MobileWallet/components/MobileSuccess.tsx

```diff
@@ -20,16 +20,17 @@ export const MobileSuccess = ({ location }: MobileSetupProps) => {
   const appDispatch = useAppDispatch()
   const { setWelcomeModal } = preferences.actions
   const [isSuccessful, setIsSuccessful] = useStateIfMounted<boolean | null>(null)
-  const { state, dispatch } = useWallet()
+  const { getAdapter, dispatch } = useWallet()
   const { vault } = location.state
 
   useEffect(() => {
     ;(async () => {
       if (!vault) return
-      const adapters = state.adapters?.get(KeyManager.Native)!
+      const adapter = await getAdapter(KeyManager.Native)
```

Comment:
i wonder if its possible to add some typscript magic such that `getAdapter` can return the correct wallet type based on `KeyManager`, negating the requirement to cast below

Reply:
We absolutely could've previously but didn't - we still could now, however, having such typings would imply importing the types (this can't be done lazily).
Assuming we had actual types and types are stripped down at runtime, I'm not sure whether or not doing so would have any incidence in packages being imported, just by us importing types?

Either way, the issue is that we currently don't have types, but rather classes, so we would have to import the actual class which would mean importing the package, so we probably want to `declare Class` for hdwallet packages as a follow-up

Reply:
Discussed this one out, will leave it for a part 4 PR in the stack since this will require an hdwallet bumpy